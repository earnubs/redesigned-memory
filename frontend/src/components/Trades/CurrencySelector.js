import React from 'react';
import PropTypes from 'prop-types';
import Select from 'react-select';

const styles = {
  container: (provided, state) => ({
    ...provided,
    width: 155,
    zIndex: state.isFocused ? 10 : 'initial',
    color: '#556468',
  }),
  control: (provided, state) => ({
    ...provided,
    borderColor: state.isFocused ? '#00bef0' : '#84959a',
    boxShadow: state.isFocused ? '0 0 0 1px #00bef0' : 'none',
    borderTopRightRadius: 0,
    borderBottomRightRadius: 0,
  }),
  singleValue: provided => ({
    ...provided,
    color: '#556468',
  }),
  indicatorSeparator: () => ({
    display: 'none',
  }),
  menu: provided => ({
    ...provided,
    zIndex: 1000000,
  }),
  menuPortal: provided => ({
    ...provided,
    zIndex: 1000000,
  }),
  menuList: provided => ({
    ...provided,
    zIndex: 1000000,
  }),
};

export default class CurrencySelector extends React.Component {

  render() {
    const {
      options,
      isLoadingCurrencies,
      handleCurrencyChange,
    } = this.props;

    return (
      <Select
        styles={styles}
        placeholder={null}
        isSearchable
        isLoading={isLoadingCurrencies}
        options={options}
        onChange={handleCurrencyChange}
        maxMenuHeight={128}
      />
    );
  }
}

CurrencySelector.defaultProps = {
  options: [],
  isLoadingCurrencies: true,
};

CurrencySelector.propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string,
    }),
  ),
  isLoadingCurrencies: PropTypes.bool,
  handleCurrencyChange: PropTypes.func.isRequired,
};
