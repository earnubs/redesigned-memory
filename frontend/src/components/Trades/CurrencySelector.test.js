import React from 'react';
import { shallow } from 'enzyme';

import CurrencySelector from './CurrencySelector';


describe('<CurrencySelector />', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<CurrencySelector handleCurrencyChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call passed onChange handler', () => {
    const spy = jest.fn();
    const wrapper = shallow(<CurrencySelector handleCurrencyChange={spy} />);
    wrapper.prop('onChange')();
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
