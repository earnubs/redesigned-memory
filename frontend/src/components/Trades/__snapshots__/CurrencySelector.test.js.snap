// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencySelector /> should render as expected 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isLoading={true}
  isSearchable={true}
  maxMenuHeight={128}
  onChange={[Function]}
  options={Array []}
  placeholder={null}
  styles={
    Object {
      "container": [Function],
      "control": [Function],
      "indicatorSeparator": [Function],
      "menu": [Function],
      "menuList": [Function],
      "menuPortal": [Function],
      "singleValue": [Function],
    }
  }
/>
`;
